lib/elastic_record/relation/search_methods.rb

```diff
@@ -236,7 +236,14 @@ def build_filter_nodes(filters, add_join_field_filter = true)
             if filter.is_a?(Arelastic::Nodes::Node)
               nodes << filter
             elsif filter.is_a?(ElasticRecord::Relation)
-              nodes << Arelastic::Queries::HasChild.new(filter.elastic_index.mapping_type, filter.as_elastic['query'])
+              join_type = if klass.in?(Array.wrap(filter.klass.try(:es_descendants)))
+                 Arelastic::Queries::HasParent
+              elsif filter.klass.in?(Array.wrap(klass.try(:es_descendants)))
+                 Arelastic::Queries::HasChild
+              else
+                raise "You can't filter a #{filter.klass.name} ('#{filter.as_elastic}') inside a #{klass.name}"
```

Comment:
`filter.as_elastic` has the potential to be a huge object. Does it really belong in the error message?

Reply:
Fixed